### PR TITLE
Task 11

### PR DESCRIPTION
**Pull request by fertant**

### CODE REVIEW

File: react-app/src/components/molecules/Dialog/DialogComponent.tsx [hunk 1]
```diff
@@ -13,11 +13,17 @@ import {
 import EditMovieForm from '../Forms/EditMovieForm';
 import AddMovieForm from '../Forms/AddMovieForm';
 import DeleteMovieForm from '../Forms/DeleteMovieForm';
+import { IMovie } from '../MovieCard/IMovie';
 
+// PATTERN: high order component
```
Comment (ygorbunkov) [inline react-app/src/components/molecules/Dialog/DialogComponent.tsx]: I may be missing something, but below doesn't seem to be a good example of HOC. I'd recommend to refer to [official doc](https://reactjs.org/docs/higher-order-components.html) on that topic.

File: react-app/src/components/molecules/MovieCard/MovieCardComponent.tsx [hunk 2]
```diff
@@ -17,9 +18,12 @@ import {
 } from './MovieCard';
 import { actionControlVisibility, actionSelectMovie } from '../../../actions/Actions';
 
+// PATTERN: render prop
```
Comment (ygorbunkov) [inline react-app/src/components/molecules/MovieCard/MovieCardComponent.tsx:21]: Aren't you suppose to pass in some dynamically rendered prop-function to implement render props?

File: react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+import {
+  actionApiFilterRequest,
+  actionSetFilter,
+  actionSelectMovieId,
+  actionControlVisibility,
+} from '../../../actions/Actions';
+import NavigationComponent from '../../molecules/Navigation/NavigationComponent';
 
+const HeaderComponent = React.lazy(() => import('../../organisms/Header/HeaderComponent'));
+const ErrorBoundary = React.lazy(() => import('../../atoms/ErrorBoundary/ErrorBoundary'));
```
Comment (ygorbunkov) [inline react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx]: I don't think it is a good idea to lazy-load `<ErrorBoundary />` component. Just think about it - what if some error will be thrown before it is loaded?

File: react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
-import { actionApiFilterRequest } from '../../../actions/Actions';
+import {
+  actionApiFilterRequest,
+  actionSetFilter,
+  actionSelectMovieId,
+  actionControlVisibility,
+} from '../../../actions/Actions';
+import NavigationComponent from '../../molecules/Navigation/NavigationComponent';
 
+const HeaderComponent = React.lazy(() => import('../../organisms/Header/HeaderComponent'));
```
Comment (ygorbunkov) [inline react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx]: You still seem to be using SSR, and `React.lazy` is [not (yet) supposed to work](https://reactjs.org/docs/code-splitting.html#reactlazy) in server-side mode. So, I'm not entirely sure how did you manage to get it working.

Comment (fertant) [reply react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx]: Replaced with loadable component

File: react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx (hunk 3 above)
Comment (fertant) [reply react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx]: React.lazy removed as it is not supported with SSR.

File: react-app/src/components/molecules/MovieCard/MovieCardComponent.tsx (hunk 2 above)
Comment (fertant) [reply react-app/src/components/molecules/MovieCard/MovieCardComponent.tsx:21]: Props item passed inside.

**Pull request merged**